tests/core/test_hookenv.py

```diff
@@ -1370,6 +1370,32 @@ def test_goal_state(self, check_output_):
         self.assertEqual(result, expect)
         check_output_.assert_called_with(['goal-state', '--format=json'])
 
+    @patch('subprocess.check_output')
+    def test_goal_state_filter_self_relations(self, check_output_):
+        data = {
+            'units': {'keystone/0': {}, 'keystone/1': {}},
+            'relations': {
+                'identity-service': {
+                    'glance/0': {},
+                    'keystone/0': {},
+                    'keystone/1': {},
+                },
+            },
+        }
+        expect = {
+            'units': {'keystone/0': {}, 'keystone/1': {}},
+            'relations': {
+                'identity-service': {
+                    'glance/0': {},
+                },
+            },
+        }
+        check_output_.return_value = json.dumps(data).encode('UTF-8')
+        result = hookenv.goal_state_filter_self_relations()
+
+        self.assertEqual(result, expect)
+        check_output_.assert_called_with(['goal-state', '--format=json'])
```

Comment:
This test isn't necessary, and doesn't belong here. The test is testing the goal_state_filter_self_relations implementation; test of the goal_state implementation should be separate.